RecoTracker/FinalTrackSelectors/plugins/DuplicateTrackMerger.cc

```diff
@@ -28,6 +32,18 @@
 
 #include "CondFormats/EgammaObjects/interface/GBRForest.h"
 
+// Having this macro reduces the need to pollute the code with
+// #ifdefs. The idea is that the condition is checked only if
+// debugging is enabled. That way the condition expression may use
+// variables that are declared only if EDM_ML_DEBUG is enabled. If it
+// is disabled, rely on the fact that LogTrace should compile to
+// no-op.
+#ifdef EDM_ML_DEBUG
+#define IfLogTrace(cond, cat) if(cond) LogTrace(cat)
+#else
+#define IfLogTrace(cond, cat) LogTrace(cat)
+#endif
```

Comment:
@Dr15Jones Configuring MessageLogger is indeed one thing (I usually replace the relevant LogDebug/LogTrace with LogWarning/LogPrint, as mostly I anyway need to look the source code to know what exactly is being done and printed). But the main point is that I want the printouts only for specific track(s)/seed(s)/hit(s)/younameit inside a loop over them.